098-using-image-volumes-to-improve-extensibility-of-Strimzi-operands.md

```diff
@@ -0,0 +1,320 @@
+      pod:
+        volumes:
+          - name: my-authorizer
+            image:
+              reference: ghcr.io/scholzj/my-authorizer:latest
+              pullPolicy: IfNotPresent
+```
+
+#### Tiered storage plugins
+
+Unlike other plugins used in Kafka brokers, the tiered storage plugin is not expected to be in the default class path.
+Instead, tiered storage has its own class path configuration in `.spec.tieredStorage.remoteStorageManager.classPath`.
+This field should be used instead of the `CLASSPATH` environment variable.
+For example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  # ...
+spec:
+  kafka:
+    # ...
+    template:
+      kafkaContainer:
+        volumeMounts:
+          - name: my-tiered-storage-plugin
+            mountPath: /mnt/my-tiered-storage-plugin/
+      pod:
+        volumes:
+          - name: my-tiered-storage-plugin
+            image:
+              reference: ghcr.io/scholzj/my-tiered-storage-plugin:latest
+              pullPolicy: IfNotPresent
+    tieredStorage:
+      type: custom
+      remoteStorageManager:
+        # ...
+        classPath: /mnt/my-tiered-storage-plugin/*
+        config:
+          # ...
+```
+
+### Immutability
+
+When using image volumes to provide Apache Kafka plugins, we do not rely anymore that all the software is provided from a single immutable container image provided by Strimzi and later modified by the user to add the custom plugin.
+Instead we compose the running software from multiple images:
+* The Strimzi image with the Strimzi software
+* One or more plugin images mounted as volumes
+
+For example, following the example Kafka Connect deployment with the configuration used in the corresponding section above:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnect
+metadata:
+  # ...
+spec:
+  # ...
+  plugins:
+    - name: echo-sink-connector
+      artifacts:
+        - type: image
+          reference: ghcr.io/scholzj/echo-sink@sha256:7078b3ccbc0d6e76fefb832dd6e4bb6b704f83611428c9d5b060ab0b6c3e8712
+          pullPolicy: Always
+```
+
+The software in this case will be defined as the following set of container images:
+* `quay.io/scholzj/kafka@sha256:fd0a060b512240c6dc45e90de26fd738fbbcd470dbb1a3c4c8ec808392db6159` as the _Strimzi_ container image from my prototype build
+* `ghcr.io/scholzj/echo-sink@sha256:7078b3ccbc0d6e76fefb832dd6e4bb6b704f83611428c9d5b060ab0b6c3e8712` as the Kafka Connect connector image
+
+Each of these are immutable container images and clearly define the software being run in a given operand.
+
+### Error handling
+
+Strimzi cannot easily detect if the `ImageVolume` feature gate is enabled and if image volumes are supported in given Kubernetes cluster.
+While we will use the documentation to warn users that this feature can be used only in Kubernetes cluster where it is supported, it might always happen that users try to use it even when not supported by Kubernetes.
```

Comment:
```suggestion
While we will use the documentation to warn users that this feature can be used only in Kubernetes cluster where it is supported, it might happen that users try to use it even when not supported by Kubernetes.
```